### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by roonle**
Missing non-default constructor, need to refactor to sample data, and add more unit test

### CODE REVIEW

**Comment by KS-D** [comment]
I will review this.

File: SecretSanta/src/SecretSanta.Business/AutomapperProfileConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace SecretSanta.Business
+{
+    public class AutomapperProfileConfiguration : Profile
+    {
+        public AutomapperProfileConfiguration()
+        {
+            CreateMap<Gift, Gift>().ForMember(property => property.Id, option => option.Ignore());
+            CreateMap<User, User>().ForMember(property => property.Id, option => option.Ignore());
+        }
```
Comment (KS-D) [inline SecretSanta/src/SecretSanta.Business/AutomapperProfileConfiguration.cs]: It looks like Group still needs to be setup with the auto-mapper.

File: SecretSanta/src/SecretSanta.Data/EntityBase.cs [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+using System.ComponentModel.DataAnnotations;
 using System.Text;
 
 namespace SecretSanta.Data
 {
     public class EntityBase
     {
-        public int Id { get; set; }
+        [Required]
+        public int Id { get; protected set; }
```
Comment (KS-D) [inline SecretSanta/src/SecretSanta.Data/EntityBase.cs]: I believe these [Required] fields are supposed to be null able.

**Comment by KS-D** [comment]
It built without errors or warning from the command line. It looks like you just have some little things fix/work on, and then some tests. Overall the code you have is pretty clean.

File: SecretSanta/test/SecretSanta.Data.Tests/GiftTests.cs [hunk 3]
```diff
@@ -15,13 +15,8 @@ public async Task Gift_CanBeSavedToDatabase()
             // Arrange
```
Comment (MarkMichaelis) [inline SecretSanta/test/SecretSanta.Data.Tests/GiftTests.cs]: I would love to have seen this class as a generic class containing more class refactored out of the other classes.

**Comment by MarkMichaelis** [comment]
# Assignment 3

## Grading

## Main Assignment

- Connect Azure DevOps up for your fork.
- Add a Status Badge (see [Add a status badge to your repository](https://docs.microsoft.com/en-us/azure/devops/pipelines/create-first-pipeline?view=azure-devops&tabs=browser%2Ctfs-2018-2#add-a-status-badge-to-your-repository)) to the `README.md` file in `master` that shows the build status on each branch. ✔
- Be sure that clicking on the badge successfully navigates one to the Azure DevOps pipeline. ✔
- Rebase to the Readme.md update on `master` to the `Assignment3` branch so it too shows the badge (be sure it points to the Assignment3 branch status) ✔
- Appropriately encapsulate Id and all fingerprint data. ✔
- Add non-default constructors to Gift and address all warnings (see BlogEngine's Post.cs file). ✔
- Add any missing entity framework CRUD unit tests for `Gift`. ✔
- Implement an Automapper Configuration between two `Gift` objects. ✔
  - Do not map the Id ✔
- Create services for (do not map primary key Id for all entities):
  - Groups ✔
  - Users ✔
  - Gifts ✔
  - When retrieving the Gift object from the service, User should be pre-populated (included) ✔
- Refactor out SampleData (see BlogEngine's SampleData.cs) ✔

## Extra Credit

- Write exceptional code that is clean, conforms to all coding guidelines, is well refactored, and easy to understand.

## Project

| ProjectName | Nullable | Solutions | Analyzers | Unexpected Disabled Warnings | Waning Level |
| -- | -- | -- | -- | -- | -- |
| SecretSanta.Api | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Business | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Data | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Web | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Api.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Business.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Data.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Web.Tests | Enable | SecretSanta.sln |  |  | 4 |


| Project Analysis |   |
| -----------------| - |
| Build/Compile | (Build succeeded: 0 Error(s), 0 Warning(s)) |
| Disabled Warnings **** | **✔**
| Nullable Enabled: | **✔** |
| Appropriately handle all warnings | **✔** |
| Test Run Succeeded: Total tests=12, Passed=12, Failed=0, Skipped=0 |

**Pull request closed without merging**